rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java

```diff
@@ -646,6 +649,172 @@ private BigInteger sendFundsToAnyAddress(
         );
     }
 
+    private BigInteger sendFundsSurpassesLockingCapToAnyAddress(
+        boolean isRskip293Active,
+        Coin lockingCapValue,
+        BtcTransactionProvider btcTransactionProvider,
+        boolean shouldTransferToContract
+    ) throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        when(activations.isActive(ConsensusRule.RSKIP134)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(isRskip293Active);
+
+        BridgeConstants bridgeConstants = spy(bridgeConstantsRegtest);
+
+        Context btcContext = mock(Context.class);
+        doReturn(bridgeConstants.getBtcParams()).when(btcContext).getParams();
+
+        Federation activeFederation = PegTestUtils.createFederation(bridgeConstants, "fa03", "fa04");
+        Federation retiringFederation = PegTestUtils.createFederation(bridgeConstants, "fa01", "fa02");
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        when(provider.getNewFederation()).thenReturn(activeFederation);
+        when(provider.getOldFederation()).thenReturn(retiringFederation);
+        when(provider.getLockingCap()).thenReturn(lockingCapValue);
+
+        ReleaseTransactionSet releaseTransactionSet = new ReleaseTransactionSet(new HashSet<>());
+        when(provider.getReleaseTransactionSet()).thenReturn(releaseTransactionSet);
+
+        Address userRefundBtcAddress = createRandomP2PKHBtcAddress(bridgeConstants.getBtcParams());
+        Address lpBtcAddress = createRandomP2PKHBtcAddress(bridgeConstants.getBtcParams());
+
+        Repository repository = createRepository();
+        // For simplicity of this test, the max rbtc value is set as the current balance for the repository
+        // This simulates that no pegin has ever been processed
+        repository.addBalance(PrecompiledContracts.BRIDGE_ADDR, co.rsk.core.Coin.fromBitcoin(bridgeConstantsRegtest.getMaxRbtc()));
+
+        BtcBlockStoreWithCache btcBlockStore = mock(BtcBlockStoreWithCache.class);
+        BtcBlockStoreWithCache.Factory mockFactory = mock(BtcBlockStoreWithCache.Factory.class);
+        when(mockFactory.newInstance(repository, bridgeConstants, provider, activations)).thenReturn(btcBlockStore);
+        Block executionBlock = Mockito.mock(Block.class);
+        when(executionBlock.getNumber()).thenReturn(10L);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withProvider(provider)
+            .withBridgeConstants(bridgeConstants)
+            .withBtcLockSenderProvider(mock(BtcLockSenderProvider.class))
```

Comment:
Just a minor comment, BridgeSupportBuilder uses a mock by default no need to set it here

Reply:
Done.